core/templates/components/summary-tile/lesson-card.component.ts

```diff
@@ -0,0 +1,152 @@
+import {StoryNode} from 'domain/story/story-node.model';
+
+interface LessonUrl {
+  classroom: string | undefined;
+  topicFragment: string | undefined;
+  story: string;
+  currentStory: StoryNode;
+}
+
+interface ThumbnailUrl {
+  filename: string;
+  id: string;
+}
+@Component({
+  selector: 'lesson-card',
+  templateUrl: './lesson-card.component.html',
+})
+export class LessonCardComponent implements OnInit {
+  @Input() story!: StorySummary | LearnerExplorationSummary | CollectionSummary;
+  @Input() topic!: string;
+
+  desc!: string;
+  imgColor!: string;
+  imgUrl!: string;
+  lessonUrl!: string;
+  progress!: number;
+  title!: string;
+  lessonTopic!: string;
+
+  constructor(
+    private urlInterpolationService: UrlInterpolationService,
+    private assetsBackendApiService: AssetsBackendApiService
+  ) {}
+
+  ngOnInit(): void {
+    if (this.story instanceof StorySummary) {
+      const completedStories = this.story.getCompletedNodeTitles().length;
+
+      this.desc = this.story.getTitle();
+      this.imgColor = this.story.getThumbnailBgColor();
+      this.imgUrl = this.getStorySummaryThumbnailUrl({
+        filename: this.story.getThumbnailFilename(),
+        id: this.story.getId(),
+      });
+
+      const nextStory =
+        completedStories === this.story.getAllNodes().length
+          ? completedStories - 1
+          : completedStories;
+
+      const lessonArgs = {
+        classroom: this.story.getClassroomUrlFragment(),
+        topicFragment: this.story.getTopicUrlFragment(),
+        currentStory: this.story.getAllNodes()[nextStory],
+        story: this.story.getUrlFragment(),
+      };
+      this.lessonUrl = this.getStorySummaryLessonUrl(lessonArgs);
+
+      this.title = `Chapter ${nextStory + 1}: ${this.story.getNodeTitles()[nextStory]}`;
+      this.progress = Math.floor(
+        (completedStories / this.story.getNodeTitles().length) * 100
+      );
+      this.lessonTopic = this.topic;
+    } else {
+      this.desc = this.story.objective;
+      this.imgColor = this.story.thumbnailBgColor;
+      this.imgUrl = this.urlInterpolationService.getStaticImageUrl(
+        this.story.thumbnailIconUrl
+      );
+      this.progress = 0;
+      this.title = this.story.title;
+
+      if (this.story instanceof CollectionSummary) {
+        this.lessonUrl = `/collection/${this.story.id}`;
+        this.lessonTopic = 'Collections';
+      } else {
+        this.lessonUrl = `/explore/${this.story.id}`;
+        this.lessonTopic = 'Community Lessons';
+      }
+    }
+  }
+
+  getProgress(arg: number): string {
+    const leftCircle = 50 >= arg ? arg : 50;
+    const rightCircle = arg > 50 ? arg - 50 : 0;
+
+    return `linear-gradient(${50 >= arg ? 180 + (arg / 100) * 180 : 270}deg, #00645c ${leftCircle}%, transparent ${leftCircle}%), linear-gradient(0deg, #00645c ${rightCircle}%, lightgray ${rightCircle}%)`;
+  }
+
+  getStorySummaryThumbnailUrl({filename, id}: ThumbnailUrl): string {
+    if (!filename) {
```

Comment:
Shouldn't this throw an exception? When a filename can be absent and what should we do in that situation?

Reply:
I don't mind returning some default image, just explain please why Lightbulb, not something else

Also, please respond on fixing comments

Reply:
In constants.ts, the "DEFAULT_THUMBNAIL_ICON" is set to Lightbulb.

Reply:
> In constants.ts, the "DEFAULT_THUMBNAIL_ICON" is set to Lightbulb

Well that's not a good approach - using _values_ of constants. If there is a constant - use it, so that when its value is changed the new value is used here.

But I still don't understand, why filename or id might be missing. Is it possible?

Reply:
It's not. I was again rewriting based on the original code. Removed, done.